src/renderer/views/Watch/Watch.js

```diff
@@ -758,10 +762,19 @@ export default Vue.extend({
         const player = this.$refs.videoPlayer.player
 
         if (player !== null && this.saveWatchedProgress) {
-          const currentTime = this.getWatchedProgress()
+          if (this.saveWatchedProgressForMusic || (!this.saveWatchedProgressForMusic && this.genre !== 'Music')) {
+            const currentTime = this.getWatchedProgress()
+            const payload = {
+              videoId: this.videoId,
+              watchProgress: currentTime
+            }
+            this.updateWatchProgress(payload)
+          }
+        }
+        if (this.genre === 'Music' && !this.saveWatchedProgressForMusic) {
           const payload = {
             videoId: this.videoId,
-            watchProgress: currentTime
+            watchProgress: 0
```

Comment:
There is no need to update the watch progress to 0 explicitly
(That shouldn't be part of this PR even it's a new intended behaviour)